### ISSUE TITLE
ToolStripComboBox causes MenuStrip to not display properly

### ISSUE THREAD

**Issue by roland5572** [dotnet/winforms#6724]
C# Code: .Net 6.0 with Visual studio 2022

```
using System;
using System.Drawing;
using System.Windows.Forms;

namespace ToolStripItemTest
{
    internal static class Program
    {
        /// <summary>
        ///  The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            ApplicationConfiguration.Initialize();
            Application.Run(CreateForm());
        }

        private static Form CreateForm()
        {
            var form = new Form()
            {
                Width = 1024,
                Height = 768,

                StartPosition = FormStartPosition.CenterScreen,
            };

            var mainMenu = new MenuStrip()
            {
                Dock = DockStyle.Top,
            };

            var menu1 = new ToolStripMenuItem("Menu 1");

            var subMenu1 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu2 = new ToolStripMenuItem("Sub Menu Item 2");
            var subMenu3 = new ToolStripMenuItem("Sub Menu Item 3");
            var subMenu4 = new ToolStripComboBox("Sub Menu Item 4") { DropDownStyle = ComboBoxStyle.DropDownList };
            var subMenu5 = new ToolStripMenuItem("Sub Menu Item 5");
            var subMenu6 = new ToolStripMenuItem("Sub Menu Item 6");
            var subMenu7 = new ToolStripMenuItem("Sub Menu Item 7");
            var subMenu8 = new ToolStripMenuItem("Sub Menu Item 8");


            menu1.DropDownItems.Add(subMenu1);
            menu1.DropDownItems.Add(subMenu2);
            menu1.DropDownItems.Add(subMenu3);
            menu1.DropDownItems.Add(subMenu4);
            menu1.DropDownItems.Add(subMenu5);
            menu1.DropDownItems.Add(subMenu6);
            menu1.DropDownItems.Add(subMenu7);
            menu1.DropDownItems.Add(subMenu8);

            var subMenu3_Sub1 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu3_Sub2 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu3_Sub3 = new ToolStripMenuItem("Sub Menu Item 1");

            subMenu3.DropDownItems.Add(subMenu3_Sub1);
            subMenu3.DropDownItems.Add(subMenu3_Sub2);
            subMenu3.DropDownItems.Add(subMenu3_Sub3);

            var subMenu6_Sub1 = new ToolStripControlHost(new Control() { Width = 200, Height = 60, BackColor = Color.LightGreen });
            subMenu6.DropDownItems.Add(subMenu6_Sub1);

            mainMenu.Items.Add(menu1);

            form.Controls.Add(mainMenu);

            return form;
        }
    }
}
```

Test Result:

Video 1:
1. Open the menu.
2. Click the combobox menu item.
3. move your mouse to other menu items.
4. Then you will find that the **Menu Hover** effect disappears completely, and when the mouse moves over other items, the menu items will no longer display the blue background and it cannot automatically expand its submenus unless you click the left mouse button. There is a confusing display situation.

https://user-images.githubusercontent.com/88176616/154834883-ed39e8b1-3687-40f7-9145-daa2378b85cb.mp4

Video 2:
1. Open the menu.
2. Move the mouse over the menu item with submenu items.
3. Move the mouse to the ComboBox menu item.
You will find that the previous menu does not close automatically and still retains the blue background.

https://user-images.githubusercontent.com/88176616/154834943-76a70376-a45f-4d5d-b61b-f9b446134a1a.mp4

**Comment by RussKie** [comment]
Did it work in .NET Framework?
________________________________
From: roland5572 ***@***.***>
Sent: Sunday, 20 February 2022 7:50 PM
To: dotnet/winforms ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [dotnet/winforms] ToolStripComboBox causes MenuStrip to not display properly (Issue #6724)


C# Code: .Net 6.0 with Visual studio 2022

using System;
using System.Drawing;
using System.Windows.Forms;

namespace ToolStripItemTest
{
    internal static class Program
    {
        /// <summary>
        ///  The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            ApplicationConfiguration.Initialize();
            Application.Run(CreateForm());
        }

        private static Form CreateForm()
        {
            var form = new Form()
            {
                Width = 1024,
                Height = 768,

                StartPosition = FormStartPosition.CenterScreen,
            };

            var mainMenu = new MenuStrip()
            {
                Dock = DockStyle.Top,
            };

            var menu1 = new ToolStripMenuItem("Menu 1");

            var subMenu1 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu2 = new ToolStripMenuItem("Sub Menu Item 2");
            var subMenu3 = new ToolStripMenuItem("Sub Menu Item 3");
            var subMenu4 = new ToolStripComboBox("Sub Menu Item 4") { DropDownStyle = ComboBoxStyle.DropDownList };
            var subMenu5 = new ToolStripMenuItem("Sub Menu Item 5");
            var subMenu6 = new ToolStripMenuItem("Sub Menu Item 6");
            var subMenu7 = new ToolStripMenuItem("Sub Menu Item 7");
            var subMenu8 = new ToolStripMenuItem("Sub Menu Item 8");


            menu1.DropDownItems.Add(subMenu1);
            menu1.DropDownItems.Add(subMenu2);
            menu1.DropDownItems.Add(subMenu3);
            menu1.DropDownItems.Add(subMenu4);
            menu1.DropDownItems.Add(subMenu5);
            menu1.DropDownItems.Add(subMenu6);
            menu1.DropDownItems.Add(subMenu7);
            menu1.DropDownItems.Add(subMenu8);

            var subMenu3_Sub1 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu3_Sub2 = new ToolStripMenuItem("Sub Menu Item 1");
            var subMenu3_Sub3 = new ToolStripMenuItem("Sub Menu Item 1");

            subMenu3.DropDownItems.Add(subMenu3_Sub1);
            subMenu3.DropDownItems.Add(subMenu3_Sub2);
            subMenu3.DropDownItems.Add(subMenu3_Sub3);

            var subMenu6_Sub1 = new ToolStripControlHost(new Control() { Width = 200, Height = 60, BackColor = Color.LightGreen });
            subMenu6.DropDownItems.Add(subMenu6_Sub1);

            mainMenu.Items.Add(menu1);

            form.Controls.Add(mainMenu);

            return form;
        }
    }
}


Test Result:

Video 1:

  1.  Open the menu.
  2.  Click the combobox menu item.
  3.  move your mouse to other menu items.
  4.  Then you will find that the Menu Hover effect disappears completely, and when the mouse moves over other items, the menu items will no longer display the blue background and it cannot automatically expand its submenus unless you click the left mouse button. There is a confusing display situation.

https://user-images.githubusercontent.com/88176616/154834883-ed39e8b1-3687-40f7-9145-daa2378b85cb.mp4<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F88176616%2F154834883-ed39e8b1-3687-40f7-9145-daa2378b85cb.mp4&data=04%7C01%7Cigor.velikorossov%40microsoft.com%7C5d4e0cd60f0f44d81be108d9f44e1a1a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637809438558752141%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=4skvKsTMkYIF5ACeSxQcKFyGT9cGz%2B7O4re6c8RFQ0E%3D&reserved=0>

Video 2:

  1.  Open the menu.
  2.  Move the mouse over the menu item with submenu items.
  3.  Move the mouse to the ComboBox menu item.
You will find that the previous menu does not close automatically and still retains the blue background.

https://user-images.githubusercontent.com/88176616/154834943-76a70376-a45f-4d5d-b61b-f9b446134a1a.mp4<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F88176616%2F154834943-76a70376-a45f-4d5d-b61b-f9b446134a1a.mp4&data=04%7C01%7Cigor.velikorossov%40microsoft.com%7C5d4e0cd60f0f44d81be108d9f44e1a1a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637809438558801073%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=hu5ayG%2Bq8xz3dIWNjuS8LP%2FDrHoWu7NXfNhJPiWmeto%3D&reserved=0>

**Comment by roland5572** [comment]
Yes, the same problem in .net framework 4.8.

**Comment by merriemcgaw** [comment]
Is there a specific use case you are trying to accomplish with this? On the surface it doesn't seem like a very common thing to need to do, so I'm curious what your use-case is.

**Comment by roland5572** [comment]
@merriemcgaw  Consistent with the interface of Viusual Studio, you can control some parameters of the interface through these controls. Just hide them in the menu bar instead of showing them directly in the interface. **Therefore, space can be saved**. Then it looks neater. I haven't tested the **ToolStripTextBox** control, not sure if it has the same problem.

https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.toolstriptextbox

![001](https://user-images.githubusercontent.com/88176616/155641299-cc147ea9-432c-4168-84fb-5d899805ad1f.png)

**Comment by RussKie** [comment]
I don't think this is a good analogy. It is common to use TSComboBoxes in toolstrips, but it is not common to have them in the dropdown menus. If the aim is to "save space", then you should consider using sub menus, because Windows users would be significantly more familiar with UX, e.g.:
![image](https://user-images.githubusercontent.com/4403806/155645974-8452df1b-dd2f-415e-9202-3b4b465d2ebc.png)

**Comment by roland5572** [comment]
Yes. That's also a good approach. But ComboBox provides a more **compact way**, the user doesn't need to move the mouse more times to select what it wants. If there are dozens of pieces of data, it may save time even more. 

In the way of programming, it is also easier. The programmer only needs to call ToolStripComboBox.Items.AddRange to create the list, without creating all the sub menu Items.
